asap/asaplib/cluster/ml_cluster_fit.py

```diff
@@ -101,83 +104,280 @@ def pack(self):
         return self.db.get_params
 
 
+# class LAIO_DB(FitClusterBase):
+#     """Laio Clustering scheme
+#     Clustering by fast search and find of density peaks
+#     https://science.sciencemag.org/content/sci/344/6191/1492.full.pdf
+#
+#     $$ \rho_i\,=\,\sum_j{\chi(d_{ij}-d_{cut})}$$ i.e. the local density of data point x_i
+#     $$ \delta_i\,=\,\min_{j:\rho_j>\rho_i}(d_{ij})$$ i.e. the minimum distance to a neighbour with higher density
+#     """
+#
+#     _pairwise = True
+#
+#     def __init__(self, deltamin=-1, rhomin=-1):
+#         self.deltamin = deltamin
+#         self.rhomin = rhomin
+#
+#     def fit(self, dmatrix, rho=None):
+#
+#         """
+#
+#         Parameters
+#         ----------
+#         dmatrix: The distance matrix of shape (Nele, Nele)
+#         rho: The log densities of the points of shape (Nele,)
+#
+#         Returns
+#         -------
+#
+#         """
+#
+#         if rho is None:
+#             raise ValueError('for fdb it is better to compute kernel density first')
+#
+#         delta, nneigh = self.estimate_delta(dmatrix, rho)
+#
+#         # I'll think about this!!! Criterion for deciding which data points act as cluster centres
+#         if self.rhomin < 0:
+#             self.rhomin = 0.2*np.mean(rho) + 0.8*np.min(rho)
+#         if self.deltamin < 0:
+#             self.deltamin = np.mean(delta)
+#
+#         plt.scatter(rho, delta)
+#         plt.plot([min(rho), max(rho)], [self.deltamin, self.deltamin], c='red')
+#         plt.plot([self.rhomin, self.rhomin], [min(delta), max(delta)], c='red')
+#         plt.xlabel('rho')
+#         plt.ylabel('delta')
+#         plt.show()
+#
+#         nclust = 0  # cluster index
+#         cl = np.zeros(len(rho), dtype='int')-1  # numpy array of -1's
+#         for i in range(len(rho)):
+#             if rho[i] > self.rhomin and delta[i] > self.deltamin:
+#                 nclust += 1
+#                 cl[i] = nclust
+#
+#         # Assignment
+#         ordrho = np.argsort(rho)[::-1]  # Indices of data points in descending order of local density
+#         rho_ord = rho[ordrho]
```

Comment:
Is **rho_ord** needed?

Reply:
I can't find a place where this **rho_ord** variable is used. Are you planning to use it in the future?

Reply:
If it's not used and isn't planned on being used I'd vote for its removal.

Reply:
Would be good to keep your old class until the point we feel confident writing it off as legacy code